### PR TITLE
Feature/909 proxy delegatecall

### LINKED ISSUE #909: Developers SBAT check contract code existence in Proxy contract
### Expected Behavior
Use `EXTCODESIZE` to check contract existence before `delegatecall` in Proxy.sol

### Current Behavior

No check

### PR DESCRIPTION
**Pull request by aaitor**
### Description

The intention of this PR is to validate if an address is a contract or EOA before calling the `delegatecall`

### Tested

Test added in proxy.ts calling _setAndInitializeImplementation with a EOA address.

### Other changes

The isContract function shouldn't be called from a constructor. Calling this function from a constructor will return 0 independently if the address given as parameter is a contract or EOA

### Related issues

Fixes #909 , #910, #911

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/protocol/contracts/common/Proxy.sol [hunk 1]
```diff
@@ -32,8 +33,17 @@ contract Proxy {
   function () external payable {
     bytes32 implementationPosition = IMPLEMENTATION_POSITION;
 
+    address implementationAddress;
+
+    assembly {
+      implementationAddress := sload(implementationPosition)
+    }
+
+	if (implementationAddress == address(0)) return;
```
Comment (asaj) [inline packages/protocol/contracts/common/Proxy.sol]: Can you add a comment here explaining why we do this?

**Review by asaj — APPROVED** [review]
Nice! LGTM so long as the end-to-end tests run (and pass)

Looks like the linter is complaining

File: packages/protocol/contracts/common/Proxy.sol (hunk 1 above)
Comment (aaitor) [reply packages/protocol/contracts/common/Proxy.sol]: :+1:

**Comment by asaj** [comment]
Hmm, looks like the end-to-end tests still did not run. @aaitor maybe you need to merge in master to pick up https://github.com/celo-org/celo-monorepo/pull/1255?

**Pull request merged**